python/cudf/cudf/core/udf/groupby_typing.py

```diff
@@ -47,6 +52,15 @@ class GroupType(numba.types.Type):
     """
 
     def __init__(self, group_scalar_type, index_type=index_default_type):
+        if (
+            group_scalar_type not in SUPPORTED_GROUPBY_NUMBA_TYPES
+            and not isinstance(group_scalar_type, types.Poison)
+        ):
+            # an frame containing an column with an unsupported dtype
+            # is calling groupby apply. Construct a GroupType with
+            # a poisoned type so we can later error if this group is
+            # used in the UDAF body
```

Comment:
Is UDAF something else? Or a typo?
```suggestion
            # used in the UDF body
```

Reply:
Yeah, "user defined aggregation function". I figured these were distinct from scalar functions. on second thought though there's no reason to introduce extra terminology here.